std/random.d

```diff
@@ -1904,17 +1904,35 @@ unittest
     import std.algorithm;
     foreach(RandomGen; PseudoRngTypes)
     {
-        auto a = [0, 1, 2, 3, 4, 5, 6, 7, 8, 9];
+        auto a = [0, 1, 1, 2, 3];
         auto b = a.dup;
         auto gen = RandomGen(unpredictableSeed);
-        partialShuffle(a, 5, gen);
-        assert(a[5 .. $] == b[5 .. $]);
-        sort(a[0 .. 5]);
-        assert(a[0 .. 5] == b[0 .. 5]);
-        partialShuffle(a, 6);
-        assert(a[6 .. $] == b[6 .. $]);
-        sort(a[0 .. 6]);
-        assert(a[0 .. 6] == b[0 .. 6]);
+
+        // NUM times, pick LEN elements from the array at random.
+        immutable int LEN = 2;
+        immutable int NUM = 750;
+        int[][] chk;
+        foreach(step; 0..NUM)
+        {
+            partialShuffle(a, LEN, gen);
+            chk ~= a[0..LEN].dup;
+        }
+
+        // Check that each possible a[0..LEN] was produced at least once.
+        // For a perfectly random RandomGen, the probability that each
+        // particular combination failed to appear would be at most
+        // 0.95 ^^ NUM which is approximately 1,962e-17.
+        // As long as hardware failure (e.g. bit flip) probability
+        // is higher, we are fine with this unittest.
+        sort(chk);
+        assert(equal(uniq(chk), [       [0,1], [0,2], [0,3],
+                                 [1,0], [1,1], [1,2], [1,3],
+                                 [2,0], [2,1],        [2,3],
+                                 [3,0], [3,1], [3,2],      ]));
+
+        // Check that all the elements are still there.
+        sort(a);
+        assert(equal(a, b));
```

Comment:
I really REALLY don't like having unit tests that are subject to random failures, even if the math says it won't happen often. Can we take some number of *hard-coded* seeds that are known to pass, and use those instead.

What I see happening if we don't fix this is PRs will randomly fail for this, and people will scratch their heads trying to figure out what they did to make this seemingly unrelated test fail. That's not a fair test.